RentACar/RentACar/CRUD/CarCRUD.cs

```diff
@@ -1,69 +1,66 @@
-﻿using System;
-using System.IO;
-using System.Text.Json;
+﻿using RentACar.Helpers;
 using System.Collections.Generic;
 using System.Linq;
-
+using RentACar.CRUD;
 
 namespace RentACar
 {
-    public class CarCRUD
+    public class CarCRUD : ICarCRUD
     {
-        private readonly string _jsonFile = @"..\..\..\CRUD\Cars.json";
+        private readonly string _jsonFile = @"..\..\..\Data\Cars.json";
 
-        private List<Car> DeserializeJsonToList(string jsonFile)
+        public Car Create(Car car)
         {
-            string existingJsonToString = File.ReadAllText(jsonFile);
-            List<Car> deserializedJson = JsonSerializer.Deserialize<List<Car>>(existingJsonToString);
+            var cars = CrudHelper.GetListFromFile(_jsonFile);
 
-            return deserializedJson;
-        }
+            car.Id = CrudHelper.GetNewId(cars);
+            cars.Add(car);
 
-        private void SerializeListToJson(List<Car> cars)
-        {
-            var options = new JsonSerializerOptions { WriteIndented = true };
+            CrudHelper.SerializeListToJson(cars, _jsonFile);
 
-            string listToString = JsonSerializer.Serialize(cars, options);
-            File.WriteAllText(_jsonFile, listToString);
+            return car;
         }
 
-        public Car Create(Car car)
+        public Car Get(int id)
         {
-            var cars = new List<Car>();
-
-            if (File.Exists(_jsonFile))
-            {
-                cars = DeserializeJsonToList(_jsonFile);
-
-                car.Id = cars.Last().Id + 1;
-                cars.Add(car);
-            }
-            else
-            {
-                car.Id = 0;
-                cars.Add(car);
-            }
+            var cars = CrudHelper.GetListFromFile(_jsonFile);
 
-            SerializeListToJson(cars);
+            var car = cars.FirstOrDefault(e => e.Id == id);
 
             return car;
         }
 
-        public Car Get(int id)
+        public Car Update(Car car)
         {
-            // Code here
-            return new Car();
+            var cars = CrudHelper.GetListFromFile(_jsonFile);
+            var carUpdated = UpdateCarInListById(cars, car);
+
+            if (carUpdated is not null) CrudHelper.SerializeListToJson(cars, _jsonFile);
+
+            return carUpdated;
         }
 
-        public Car Update(Car car)
+        public void Delete(int id)
         {
-            // Code here
-            return new Car();
+            var cars = CrudHelper.GetListFromFile(_jsonFile);
+            cars.Remove(cars.FirstOrDefault(car => car.Id == id));
+            CrudHelper.SerializeListToJson(cars, _jsonFile);
         }
 
-        public void Delete(int id)
+        private static Car UpdateCarInListById(List<Car> cars, Car car)
```

Comment:
I'm 99.9% sure you made this method static because VS suggested to, but I really don't like static methods. Being honest, there's no real reason to change tho, just a little rant 😛

Reply:
Honestly, i copied it from the static helper class and I didn't pay attention 😳 (which is worse than accepting what is suggested by Visual Studio 🤣)
By curiosity, why you don't like that? In this case, is it because it is not a static class?

Reply:
> By curiosity, why you don't like that? In this case, is it because it is not a static class?

exactly